static/app/actionCreators/prompts.tsx

```diff
@@ -135,6 +140,164 @@ export function usePromptsCheck(
   );
 }
 
+/**
+ * Get the status of many prompts in a single query
+ */
+export function usePrompts({
+  features,
+  organization,
+  projectId,
+  daysToSnooze,
+  options,
+}: {
+  features: string[];
+  organization: Organization | null;
+  daysToSnooze?: number;
+  options?: Partial<UseApiQueryOptions<PromptResponse>>;
+  projectId?: string;
+}) {
+  const api = useApi({persistInFlight: true});
+  const prompts = usePromptsCheck({feature: features, organization, projectId}, options);
+  const queryClient = useQueryClient();
+  const isPromptDismissed: Record<string, boolean> = useMemo(() => {
+    if (prompts.isSuccess) {
+      return features.reduce(
+        (acc, feature) => {
+          const prompt = prompts.data.features?.[feature];
+          acc[feature] = promptIsDismissed(
+            {dismissedTime: prompt?.dismissed_ts, snoozedTime: prompt?.snoozed_ts},
+            daysToSnooze
+          );
+          return acc;
+        },
+        {} as Record<string, boolean>
+      );
+    }
+    return {};
+  }, [prompts.isSuccess, prompts.data?.features, features, daysToSnooze]);
+
+  const getCachedData = useCallback(() => {
+    // this is necessary in the case multiple prompts need to be updated in sequence
+    return getApiQueryData<PromptResponse>(
+      queryClient,
+      makePromptsCheckQueryKey({organization, feature: features, projectId})
+    );
+  }, [queryClient, organization, features, projectId]);
+
+  const dismissPrompt = useCallback(
+    (feature: string) => {
+      if (!organization) {
+        return;
+      }
+      promptsUpdate(api, {
+        organization,
+        projectId,
+        feature,
+        status: 'dismissed',
+      });
+
+      const existingData = getCachedData();
+
+      // Update cached query data
+      // Will set prompt to dismissed
+      setApiQueryData<PromptResponse>(
+        queryClient,
+        makePromptsCheckQueryKey({
+          organization,
+          feature: features,
+          projectId,
+        }),
+        () => {
+          const dismissedTs = Date.now() / 1000;
+          return {
+            data: {dismissed_ts: dismissedTs},
+            features: {...existingData?.features, [feature]: {dismissed_ts: dismissedTs}},
```

Comment:
`existingData` can be out of date so it is better to use the cached data that gets passed in as an argument to this function

Reply:
that was my intent in adding `getCachedData`

what was happening was that i'd dismiss the alert which then calls `dismissPrompt` on each of the feature prompts, without `getCachedData` each subsequent update to the cache was overwriting the previous one

Reply:
Hmm that seems weird since that's exactly use use case of the argument. You were doing something like this?

```tsx
setApiQueryData<PromptResponse>(
  queryClient,
  makePromptsCheckQueryKey(...),
  (existingData) => {
    const snoozedTs = Date.now() / 1000;
    return {
      data: {snoozed_ts: snoozedTs},
      features: {...existingData?.features, [feature]: {snoozed_ts: snoozedTs}},
    };
  }
)

```

Reply:
OH i see i thought you were referencing the argument to the `dismissPrompt` function, this makes a lot more sense